.github/workflows/test-and-publish.yml

```diff
@@ -0,0 +1,47 @@
+# Github actions that run tests on each commit, and publish to PyPi on tagged releases
+# build using https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+
+name: Test and Release to PyPi
+
+on:
+  push
+
+jobs:
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
+    - name: Install and build
+      run: >-
+        python -m
+        pip install ".[dev]"
+        build
+        --user
+    - name: build binary and tarball
+      run: >-
+        python -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        .
+    - name: test
+      run: >-
+        python -m
+        pip install --upgrade pytest
+        pytest
+    - name: Publish distribution to test pypi
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
```

Comment:
This will publish to test PyPI on every push. I don't think that's what we want, right? Rather, we can use test PyPI to validate that this action is working before switching to the production PyPI endpoint.

Reply:
My thought process here was that we publish to test PyPi on each push, that way we know on push whether we can tag a commit to release changes to prod PyPi. I thought this would be fine, but we can always consider other options.

Would it make more sense to only push to PyPi test and prod on tag changes, but for different tag names? For ex, maybe only push to PyPi test on tags that begin with "test" and only push to PyPi prod on tags beginning with "v".

Or is the idea to just push to PyPi test and prod sequentially on the same "marker" (i.e. tag)?

Reply:
Ok, I think this is fine. This means test PyPI will have the bleeding edge version from master, and prod PyPI will have the stable releases. That makes sense.